core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/io/writeJson.kt

```diff
@@ -267,3 +274,29 @@ internal fun KlaxonJson.encodeDataFrameWithMetadata(
         ),
     )
 }
+
+/**
+ * Turn a [DataFrameSchema] into a datastructure that can be parsed
+ * to a JSON serializer
+ *
+ * Each column is represented the following way:
+ * - value columns: `{name: "<columnName>", kind: "ValueColumn", type: "<kType>" }`
+ * - colum groups: `{name: "<groupNameName>", kind: "ColumnGroup", group: [<GroupMember>,...] }`
+ * - data frames: `{name: "<columnName>", kind: "FrameColumn", dataframe: [<InnerDataFrameSchema>,...] }`
+ */
+internal fun DataFrameSchema.toJson(): MutableList<MutableMap<String, Any?>> {
+    val list: MutableList<MutableMap<String, Any?>> = mutableListOf()
+    columns.forEach { (name: String, columnSchema: ColumnSchema) ->
+        val schemaData = mutableMapOf<String, Any?>()
+        schemaData["name"] = name
+        schemaData["kind"] = columnSchema.kind.toString()
+        when (columnSchema) {
+            is ColumnSchema.Value -> schemaData["type"] = columnSchema.type.toString()
```

Comment:
We also have a function to turn KType to String, it's used in HTML rendering and DataFrameSchema.toString

https://github.com/Kotlin/dataframe/blob/6be5e1ac9a5b2bac84fed690740adc1ebcc1d22c/core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/Rendering.kt#L51

What do you think? Suits for AI actions?

Reply:
Looking into that method, it seems to hide some of the type information in some cases. So I do not think it is suitable for AI Actions. At least if we want to be as specific as possible with the context information. Also, if others want to use the type information, we probably need the fully qualified type as well

Reply:
Looking into this method, it seems to remove type information in some cases, which might make it problematic when we want to use the metadata to improve the context of AI actions. So for now I would prefer to keep the fully qualified names that also include nullability, e.g. `Kotlin.String?`